lib/MentionsForm/index.js

```diff
@@ -82,10 +86,10 @@ class MentionsForm extends Component {
     this.setState({ inputValue: '' });
   };
 
-  updateInputValue = e => {
-    this.props.onInputChange(e.target.value);
-    this.setState({ inputValue: e.target.value });
-  };
+  updateInputValue = e =>
+    this.setState({ inputValue: e.target.value }, () =>
+      this.props.onInputChange(e.target.value)
```

Comment:
Why do we need to do this in the `setState` callback?

Reply:
I think to ensure that we only call `onInputChange` after the value has actually changed

Reply:
Why do we need to do that?

Reply:
Its probably not essential, its just 'safer'. If you're triggering something after a value has changed you're making sure that thats actually the case. Since we're just passing the event value its even less needed though. I think I just added it in as a long shot help with my race condition business.

Just curious, is there a downside to doing this?

Reply:
Only downside I can see is extra code and that may not be doing anything and potential time spent for the next dev trying to figure out why it's like this (not usual pattern I don't think).

I will leave to you though, I'm happy you have explained the reasoning behind it :+1:

Reply:
Ah ok cool, I've taken it out since there's no real reason for it to be there... For some reason I had it in my mind that it was liked that before I touched it in this PR